Applications/WebPortal/WebApp/library/source/pages/dashboard_page/dashboard_controller.tsx

```diff
@@ -115,6 +116,18 @@ export class DashboardController extends React.Component<Properties, State> {
         displaySize={this.props.displaySize}/>);
   }
 
+  private renderGroupPage = () => {
+    const pattern = Path.pathToRegexp('/group/:id(\\d+)?', [], { end: false });
```

Comment:
I'm not sure what's being asked. Should I move the id handling from the Path to a separate private static method?

Reply:
The expression ```Path.pathToRegexp('/group/:id(\\d+)?', [], { end: false });``` is being evaluated on every single call to ```renderGroupPage``` yielding the exact same value on every evaluation. Instead of performing that evaluation on every call to ```renderGroupPage```, the suggestion is to evaluate the expression once on the first invocation of ```renderGroupPage``` and to reuse the result of that evaluation on every successive call. To do this, you declare ```pattern``` to be a ```static``` variable:

    static const pattern = Path.pathToRegexp('/group/:id(\\d+)?', [], { end: false });